scripts/test/Muon/run_string_utils_test.py

```diff
@@ -0,0 +1,194 @@
+        for invalid_string in invalid_strings:
+            self.assertEqual(utils.validate_run_string(invalid_string), False)
+
+    def test_validate_run_string_returns_false_for_non_run_strings(self):
+        invalid_strings = ["1-5a", "abc", "a-z", "a,b", "!!", "\\", "@##"]
+        for invalid_string in invalid_strings:
+            self.assertEqual(utils.validate_run_string(invalid_string), False)
+
+    def test_run_string_to_list_correct_for_consecutive_runs(self):
+        run_strings = ["1-5", "1,2,3,4,5"]
+        for run_string in run_strings:
+            run_list = utils.run_string_to_list(run_string)
+            self.assertEqual(run_list, [1, 2, 3, 4, 5])
+
+    def test_run_string_to_list_sorts_list_ascending(self):
+        run_strings = ["5,4,3,2,1", "4,5,1-3"]
+        for run_string in run_strings:
+            run_list = utils.run_string_to_list(run_string)
+            self.assertEqual(run_list, [1, 2, 3, 4, 5])
+
+    def test_run_string_to_list_removes_duplicates(self):
+        run_strings = ["5,5,5,5,4,3,2,1,1", "1,2,4,5,1-3,1-5,1-4"]
+        for run_string in run_strings:
+            run_list = utils.run_string_to_list(run_string)
+            self.assertEqual(run_list, [1, 2, 3, 4, 5])
+
+    def test_run_string_to_list_throws_for_incorrectly_placed_range_separator(self):
+        run_strings = ["-1,2,3", "1,2,3-"]
+        for run_string in run_strings:
+            with self.assertRaises(IndexError) as context:
+                utils.run_string_to_list(run_string)
+                self.assertTrue(run_string + " is not a valid run string" in context.exception)
+
+    def test_run_string_to_list_throws_for_incorrectly_placed_delimiter(self):
+        run_strings = [",1,2,3", "1,2,3,"]
+        for run_string in run_strings:
+            with self.assertRaises(IndexError) as context:
+                utils.run_string_to_list(run_string)
+                self.assertTrue(run_string + " is not a valid run string" in context.exception)
+
+    def test_run_string_to_list_handles_non_consecutive_runs(self):
+        run_lists = [[1, 3, 4, 5], [0, 1, 2, 3, 8, 9, 199, 200]]
+        run_strings = ["1,3-5", "0-3,8-9,199-200"]
+        for i in range(len(run_lists)):
+            self.assertEqual(utils.run_string_to_list(run_strings[i]), run_lists[i])
+
+    def test_run_string_to_list_orders_non_consecutive_runs(self):
+        run_string = "1-3,48-50"
+        run_list = [1, 2, 3, 48, 49, 50]
+        self.assertEqual(utils.run_string_to_list(run_string), run_list)
+
+
+class RunStringUtilsIncrementAndDecrementRunLists(unittest.TestCase):
+
+    def test_cannot_increment_or_decrement_by_zero_or_negative_numbers(self):
+        run_list = [1, 2, 3, 4, 5]
+        self.assertEqual(utils.increment_run_list(run_list, 0), run_list)
+        self.assertEqual(utils.increment_run_list(run_list, -1), run_list)
+        self.assertEqual(utils.decrement_run_list(run_list, 0), run_list)
```

Comment:
lines 119 and 120 are the same as the 2 above. So I do not see what value they add

Reply:
Not quite, the first two use `increment_run_list`, the second two use `decrement_run_list`